client/my-sites/earn/components/add-edit-plan-modal/index.tsx

```diff
@@ -40,46 +43,25 @@ type DefaultNames = {
 	[ key: string ]: string;
 };
 
-const STRIPE_MINIMUM_CURRENCY_AMOUNT: StripeMinimumCurrencyAmounts = {
-	USD: 0.5,
-	AUD: 0.5,
-	BRL: 0.5,
-	CAD: 0.5,
-	CHF: 0.5,
-	DKK: 2.5,
-	EUR: 0.5,
-	GBP: 0.3,
-	HKD: 4.0,
-	INR: 0.5,
-	JPY: 50,
-	MXN: 10,
-	NOK: 3.0,
-	NZD: 0.5,
-	PLN: 2.0,
-	SEK: 3.0,
-	SGD: 0.5,
-};
-
-const currencyList = Object.keys( STRIPE_MINIMUM_CURRENCY_AMOUNT ).map( ( code ) => ( { code } ) );
-
 /**
  * Return the minimum transaction amount for a currency.
  * If the defaultCurrency is not the same as the current currency, return the double, in order to prevent issues with Stripe minimum amounts
  * See https://wp.me/p81Rsd-1hN
  *
+ * @param {StripeMinimumCurrencyAmounts} currency_min - Minimum transaction amount for each currency.
  * @param {string} currency - Currency.
  * @param {string} connectedAccountDefaultCurrency - Default currency of the current account
  * @returns {number} Minimum transaction amount for given currency.
  */
 function minimumCurrencyTransactionAmount(
+	currency_min: StripeMinimumCurrencyAmounts,
 	currency: string,
 	connectedAccountDefaultCurrency: string
 ): number {
-	if ( connectedAccountDefaultCurrency === currency.toUpperCase() ) {
-		return STRIPE_MINIMUM_CURRENCY_AMOUNT[ currency ];
+	if ( connectedAccountDefaultCurrency.toUpperCase() === currency ) {
```

Comment:
```suggestion
	if ( connectedAccountDefaultCurrency === currency ) {
```

Seing the old check,. I don't think the toUpperCase is needed here? 🤔

Reply:
I logged these out before the `if` check without any modifiers. It seems that `connectedAccountDefaultCurrency` does need to be uppercased.
![Screen Shot 2023-09-03 at 11 35 18 PM](https://github.com/Automattic/wp-calypso/assets/8002138/f9e35cce-bb7d-43bc-990b-44163c495ada)